packages/salesforcedx-vscode-replay-debugger/src/index.ts

```diff
@@ -107,6 +114,49 @@ function getDialogStartingPath(): vscode.Uri | undefined {
   }
 }
 
+async function retrieveLineBreakpointInfo(): Promise<boolean> {
+  const sfdxApex = vscode.extensions.getExtension(
+    'salesforce.salesforcedx-vscode-apex'
+  );
+  if (sfdxApex && sfdxApex.exports) {
+    let expired = false;
+    let i = 0;
+    while (!sfdxApex.exports.isLanguageClientReady() && !expired) {
+      await imposeSlightDelay(100);
```

Comment:
@JimSuplizio - What is the root cause for needing this? This seems brittle. Is this a temporary solution that got checked in by mistake?

CC: @DatGuyJonathan , @lcampos

Reply:
@vazexqi This was necessary because the language client may not actually be ready. The replay extension and the replay DA request source/line information at different times. When checkpoints were added we needed the source/line information much sooner than the DA, which only needs it when it starts replay-debugging. The problem here is that the extension was trying to invoke getLineBreakpointInfo when the extension wasn't ready and failing. The difference here was less than a couple hundred milliseconds. The extensions, themselves, have an onReady event that signals when they're ready (look at the changes for packages/salesforcedx-vscode-apex/src/index.ts, in this review), for that event I set the flag. In the replay debugger extension we will wait up to 3 seconds in 100 millisecond intervals to check the status. It either expires and reports an error that we're unable to get source line information from the language server OR the extension is ready and we can call to get the source line information.